hasher-matcher-actioner/scripts/soak_test_system

```diff
@@ -0,0 +1,115 @@
+#! /usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Soak/endurance test a deployed instance of HMA
+
+Besides deploying what is needed:
+- A Cognito User that has access to the deloyed system
+- Local AWS certs (if you were able to run tf apply you should be fine)  
+- A place to set off the test (ec2 instance + long running tmux session)
+
+Structure of the soak test
+
+Initial set up (no point in a long running test if this doesn't work)
+- Access to API via refresh token (+ client_id) 
+- Image used has a hash that will match the systems index
+- PR Configs (and associated signals) exist so match records will be created
+- Action Rules (and associated actions) exist so action will send post request
+
+Run
+- Submit Content Test (done every interval=seconds)
+- Sleep
+- Refresh Token
+- Repeat
+
+ToDo Missing more involed checks (between submit and webhook):
+  - The content record was created (with the expected metadata/additional fields) 
+  - Brief wait: hash was created of that image and it matches the expected
+  - wait: match record was created (if expected) and maps to correct signal values
+  - longer wait: action record was created and maps to correct action label
+
+"""
+
+
+import argparse
+import time
+
+from submit_content_test import DeployedInstanceTestHelper
+
+
+# Defaults (often it is easier to edit the script than provide the args)
+API_URL = ""
+REFRESH_TOKEN = ""
+CLIENT_ID = ""
+
+LISTENER_EXTERNAL_HOSTNAME = ""
+LISTENER_PORT = 8080
+
+
+def start_soak(
+    api_url: str,
+    refresh_token: str,
+    client_id: str,
+    hostname: str,
+    port: int,
+    interval: int,
+):
+    helper = DeployedInstanceTestHelper(api_url, "", client_id, refresh_token)
+    helper.refresh_api_token()
+    helper.set_up_basic_test(hostname, port)
+    counter = 0
+    while True:
+        print(f"Beginning Submit Test:{counter}")
+        helper.run_basic_test_with_webhook_listener(hostname, port)
```

Comment:
Dunno whether you wanted to add a "load" angle to this PR or not, but here are some things to consider:

* If in a single run, we submit an image _AND_ wait for its action to be called, it will take all queue timeouts (4*30) + wait time (10) = 130 seconds.
* This would mean 0.5 queries per minute which would not be meaningful load.

Can we decouple the 'submission' from the 'counting'? Make that async?
- start the mock server
- load at configurable rate, count total number of requests
- Ctrl-C or configured number of requests
- Poll the mock server until number of requests is reached or timeout at 130 seconds (from above)

This allows you to scale up the 'load' phase and makes for meaningful loads.